dash/_hooks.py

```diff
@@ -0,0 +1,93 @@
+import typing as _t
+
+_ns = {
+    "setup": [],
+    "layout": [],
+    "routes": [],
+    "error": [],
+    "callback": [],
+}
+
+
+def layout(func):
+    """
+    Run a function when serving the layout, the return value
+    will be used as the layout.
+    """
+    _ns["layout"].append(func)
+    return func
+
+
+def setup(func):
+    """
+    Can be used to get a reference to the app after it is instantiated.
+    """
+    _ns["setup"].append(func)
+    return func
+
+
+def route(name=None, methods=("GET",)):
+    """
+    Add a route to the Dash server.
+    """
+
+    def wrap(func):
+        _name = name or func.__name__
+        _ns["routes"].append((_name, func, methods))
+        return func
+
+    return wrap
+
+
+def error(func: _t.Callable[[Exception], _t.Any]):
```

Comment:
you're declaring the type of `func` here but you don't declare types in the previous registration functions - declare them all for consistency?

Reply:
There is some issue with cyclic dependencies with the other types, I'll see what I can do.